### DISCUSSION TITLE
Is the Zero 2 W supported?

### DISCUSSION THREAD

**Discussion by Bastczuak** [rust-embedded/rust-raspberrypi-OS-tutorials#162] [General]
Hello,

so the PI's are really scarce right now here in good ol Germany. I have only a Rasperry PI 2 W. Am I wrong or could I use this instead of the 3? Because they have the same Cortex-A53 64-bit, right?

**Comment by andre-richter** [comment]
Never tested it on one, but looks like it should work, at least CPU wise. I don’t know if they changed anything in the address map.

It seems to have only 512 MiB DRAM. From the top of my head I can’t remember right now if I made any assumptions about DRAM size somewhere.

Give it a try with the rpi3 BSP.

**Reply by Bastczuak** [reply]
Thanks I give it a try. Maybe I will update with my experience later.

**Comment by baoyuxu** [comment]
Hello.
Did you succeed with Zwro 2w?

**Comment by BartMassey** [comment]
I tried chapter 05 on the Pi Zero 2 W with the `rpi3` BSP and the recommended serial board. Serial board worked fine with Pi OS Lite, but no serial output for Tutorial. Should receive a Pi 3B+ tomorrow and will verify that things work there. Then will start to try to figure out what is needed for Pi Zero 2 W.

**Reply by h9j6k** [reply]
Any update is greatly appreciated. I am reading tutorial with a pi zero 2w.

> I tried chapter 05 on the Pi Zero 2 W with the `rpi3` BSP and the recommended serial board. Serial board worked fine with Pi OS Lite, but no serial output for Tutorial. Should receive a Pi 3B+ tomorrow and will verify that things work there. Then will start to try to figure out what is needed for Pi Zero 2 W.

**Reply by BartMassey** [reply]
I got the tutorial to work with the Pi Zero 2 W. Here's some instructions:

-----

The partition that came with my SDXC card was not good enough. Needed to remove partition 1 and replace it with one created by parted with

          mkpart primary fat32 32768s  3905535s

This is about a 2GB partition, which should be more than enough.

Having replaced the partition, now initialized it with parted via

         mkfs.fat -n BOOT -F 32

It might be better to just create a Linux boot disk for the Pi and start with Partition 1 from that. Remove all the unneeded stuff and should be good to go.

I've had a lot of trouble on Linux with Docker not mapping my `/dev/USB0`. Not sure what's up there. The Ruby `miniterm` being used by the examples does not seem to run with modern Ruby on my Debian box, so Docker is still needed. I worked around everything by using `minicom` instead, but that's hard mode…

Let me know what you get to.
